pkg/docker/cog.py

```diff
@@ -504,6 +504,11 @@ def validate_and_convert_inputs(
     return inputs
 
 
+def run_model(model, inputs, cleanup_functions):
+    result = model.run(**inputs)
+    if isinstance(result, Path):
+        cleanup_functions.append(lambda: shutil.rmtree(result))
+    return result
```

Comment:
Not important, but the passing by reference took some mental gymnastics to figure out what was going on. Maybe a clearer implementation would be to return `result, cleanup_function` where the latter is either a function or `None`.

Reply:
I added documentation to this function.